### PR TITLE
Admin Page: Remove direct access to window.Initial_State from connection-settings component

### PR DESCRIPTION
**Pull request by oskosk**
#### Testing instructions

As an Admin and with the React Dev tools open, check that the `ConnectionSettings` component is receiving at least these props and that they have truthy values. 

* userCanDisconnectSite
* userIsMaster

And these other three should have values according to your  WordPress.com user and your local username

* userWpComLogin
* userWpComEmail
* username

### CODE REVIEW

File: _inc/client/state/initial-state/reducer.js [hunk 1]
```diff
@@ -51,6 +51,22 @@ export function userCanDisconnectSite( state ) {
 	return get( state.jetpack.initialState.userData.currentUser.permissions, 'disconnect', false );
 }
 
+export function userIsMaster( state ) {
+	return get( state.jetpack.initialState.userData.currentUser, 'isMaster', false );
+}
+
+export function getUserWpComLogin( state ) {
+	return get( state.jetpack.initialState.userData.currentUser, [ 'wpcomUser', 'login'] );
```
Comment (dereksmart) [inline _inc/client/state/initial-state/reducer.js]: Inconsistent spacing in the last bracket

File: _inc/client/state/initial-state/reducer.js [hunk 2]
```diff
@@ -51,6 +51,22 @@ export function userCanDisconnectSite( state ) {
...
+}
+
+export function getUserWpComEmail( state ) {
+	return get( state.jetpack.initialState.userData.currentUser, [ 'wpcomUser', 'email'] );
```
Comment (dereksmart) [inline _inc/client/state/initial-state/reducer.js]: Inconsistent spacing in the last bracket

File: _inc/client/state/initial-state/reducer.js [hunk 3]
```diff
@@ -51,6 +51,22 @@ export function userCanDisconnectSite( state ) {
...
+}
+
+export function getUsername( state ) {
+	return get( state.jetpack.initialState.userData.currentUser, [ 'username'] );
```
Comment (dereksmart) [inline _inc/client/state/initial-state/reducer.js]: Inconsistent spacing in the last bracket

**Comment by dereksmart** [comment]
LGTM after spacing!

File: _inc/client/state/initial-state/reducer.js (hunk 2 above)
Comment (oskosk) [inline _inc/client/state/initial-state/reducer.js]: They were consistently wrong in the 4 places though. That adds some value

**Pull request merged**